app/app/src/main/kotlin/com/hedvig/app/feature/loggedin/ui/LoggedInActivity.kt

```diff
@@ -175,25 +178,29 @@ class LoggedInActivity : AppCompatActivity() {
       )
       val darkTheme = hedvigAppState.darkTheme
       EnableEdgeToEdgeSideEffect(darkTheme)
-      HedvigTheme(darkTheme = darkTheme) {
-        HedvigApp(
-          hedvigAppState = hedvigAppState,
-          hedvigDeepLinkContainer = hedvigDeepLinkContainer,
-          activityNavigator = activityNavigator,
-          getInitialTab = {
-            intent.extras?.getString(INITIAL_TAB)?.let {
-              TopLevelGraph.fromName(it)
-            }
-          },
-          clearInitialTab = {
-            intent.removeExtra(INITIAL_TAB)
-          },
-          shouldShowRequestPermissionRationale = ::shouldShowRequestPermissionRationale,
-          market = market,
-          imageLoader = imageLoader,
-          languageService = languageService,
-          hedvigBuildConstants = hedvigBuildConstants,
-        )
+      CompositionLocalProvider(
```

Comment:
and here I understood that I in fact do not fully understand the usage of CompositionLocalProvider :)) probably need to talk about that

Reply:
Good idea, let's talk about it on Thursday.
In the meantime interestingly the way I learned about this like 3 years ago was by actually looking at what React Context is https://legacy.reactjs.org/docs/context.html. It's more or less the same concept 😅

I see that by now there are some nice docs https://developer.android.com/jetpack/compose/compositionlocal for compose too.
The gist is that we want to avoid using those in most cases, since they bring in implicit dependencies on this local being present but there's no guarantee that it's gonna be present and if it's not it's a runtime crash.

Here I felt like using it still since it's a composition local that is provided by Android already by default, we are only enchasing its capabilities with this change.

And by avoiding using those most of the time we've seen that problem before with some of our previews failing to work because some permission APIs are internally using a composition local which assumes some things that are not true for the previews. It's a super super easy API to misuse and bring maintainability problems, so that's why we do not use it for anything else besides what is already provided by the system (LocalUriHandler, LocalContext, LocalResources etc.) and what the m3 library wants in order to work (MaterialTheme.colorScheme... is also working off of composition locals)

Reply:
Great, I'll read it up by Thursday then